File: mobile/lib/shared/models/order_stage.dart
/// Logistics order stages (aura-server `order.stage`), in pipeline order.
enum OrderStage {
  approved,
  sourcing,
  inRoute,
  delivered,
  installed,
  returned;

  static OrderStage fromApi(String? stage) {
    switch (stage) {
      case 'sourcing':
        return OrderStage.sourcing;
      case 'in_route':
        return OrderStage.inRoute;
      case 'delivered':
        return OrderStage.delivered;
      case 'installed':
        return OrderStage.installed;
      case 'returned':
        return OrderStage.returned;
      case 'approved':
      default:
        return OrderStage.approved;
    }
  }

  String get apiValue {
    switch (this) {
      case OrderStage.inRoute:
        return 'in_route';
      default:
        return name;
    }
  }

  String get label {
    switch (this) {
      case OrderStage.approved:
        return 'Pedido';
      case OrderStage.sourcing:
        return 'Separação';
      case OrderStage.inRoute:
        return 'Rota';
      case OrderStage.delivered:
        return 'Entregue';
      case OrderStage.installed:
        return 'Instalado';
      case OrderStage.returned:
        return 'Reversa';
    }
  }
}

File: backend-spring/src/main/java/br/com/fiap/aura/domain/enums/OrderStage.java
package br.com.fiap.aura.domain.enums;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/** Máquina de estados do pedido: approved → sourcing → in_route → delivered → installed → returned. */
public enum OrderStage {
    APPROVED("approved"),
    SOURCING("sourcing"),
    IN_ROUTE("in_route"),
    DELIVERED("delivered"),
    INSTALLED("installed"),
    RETURNED("returned");

    private final String wire;

    OrderStage(String wire) {
        this.wire = wire;
    }

    @JsonValue
    public String value() {
        return wire;
    }

    @JsonCreator
    public static OrderStage from(String raw) {
        for (OrderStage s : values()) {
            if (s.wire.equalsIgnoreCase(raw)) {
                return s;
            }
        }
        throw new IllegalArgumentException("Estágio desconhecido: " + raw);
    }

    /** Próximo estágio na cadeia, ou vazio quando terminal (returned). */
    public OrderStage next() {
        return switch (this) {
            case APPROVED -> SOURCING;
            case SOURCING -> IN_ROUTE;
            case IN_ROUTE -> DELIVERED;
            case DELIVERED -> INSTALLED;
            case INSTALLED -> RETURNED;
            case RETURNED -> null;
        };
    }
}
